### PR TITLE
Session persist using cookie

### PR DESCRIPTION
**Pull request by theo333**

### CODE REVIEW

File: client/components/App.js [hunk 1]
```diff
@@ -1,51 +1,67 @@
...
 import { HashRouter as Router, Route, Switch } from 'react-router-dom';
+import Cookies from 'js-cookie';
 import Menu from './Menu';
 import Login from './Login';
 import ProductSingle from './ProductSingle';
 import Nav from './Nav';
 import Signup from './Signup';
 import Home from './Home';
 import Order from './Order';
-import Checkout from './Checkout';
```
Comment (alexwaite) [inline client/components/App.js]: Shouldn't we keep the import of the Checkout component?

File: client/components/App.js [hunk 2]
```diff
@@ -1,51 +1,67 @@
...
-      </header>
-      <section id="content" className="container-fluid">
-        <Switch>
-          <Route exact path="/" component={Home} />
-          <Route exact path="/menu" component={Menu} />
-          <Route path="/menu/product/:productId" component={ProductSingle} />
-          <Route path="/login" component={Login} />
-          <Route path="/signup" component={Signup} />
-          <Route path="/bag" component={Order} />
-          <Route path="/checkout" component={Checkout} />
```
Comment (alexwaite) [inline client/components/App.js:36]: Shouldn't we keep the Route for Checkout?

File: server/api/auth.js [hunk 3]
```diff
@@ -15,20 +16,20 @@ router.delete('/logout', (req, res) => {
...
 // });
 
-// const userNotFound = next => {
-//     const err = new Error('User not found')
-//     err.tatus = 404
-//     next(err)
-// }
+const userNotFound = next => {
+  const err = new Error('User not found');
+  err.tatus = 404;
```
Comment (alexwaite) [inline server/api/auth.js]:
```suggestion
  err.status = 404;
```

**Review by alexwaite — CHANGES_REQUESTED** [review]
Good job getting the cookie and session functionality up and running! We can build on this to have a persistent cart and user on state.

File: client/components/App.js (hunk 1 above)
Comment (theo333) [reply client/components/App.js]: Yes, something happened during merge that messed things up.  Fixing ...

File: client/components/App.js (hunk 2 above)
Comment (theo333) [reply client/components/App.js:36]: Yes, something happened during merge that messed things up.  Fixing ...

File: server/api/auth.js (hunk 3 above)
Comment (alexwaite) [inline server/api/auth.js]:
```suggestion
  err.status = 404;
```

**Review by alexwaite — APPROVED** [review]

**Pull request merged**